File: leetcode-submissions/1394_Find_Lucky_Integer_in_an_Array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map<int,int>frequency;
        for(int ele:arr){
            frequency[ele]++;
        }
        int ans=-1;
        for(auto ele:frequency){
            if(ele.first==ele.second){
                ans=max(ele.first,ans);
            }
        }
        return ans;
    }
};

File: leetcode-submissions/1394_Find_Lucky_Integer_in_an_Array.ts
function findLucky(arr: number[]): number {
    const mp:Map<number , number> = new Map();
    for(const num of arr){
        mp.set(num , (mp.get(num)! || 0)+1);
    }
    let maxAns : number = Number.MIN_SAFE_INTEGER;
    for(const [key , value] of mp.entries()){
        if(key===value){
            maxAns = Math.max(maxAns , key);
        }
    }
    return (maxAns===Number.MIN_SAFE_INTEGER)?-1:maxAns  ;
};